controller_manager/src/resource_manager.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <utility>
+
+#include "hardware_interface/components/actuator.hpp"
+#include "hardware_interface/components/actuator_interface.hpp"
+#include "hardware_interface/components/sensor.hpp"
+#include "hardware_interface/components/sensor_interface.hpp"
+#include "hardware_interface/components/system.hpp"
+#include "hardware_interface/components/system_interface.hpp"
+#include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_info.hpp"
+
+#include "pluginlib/class_loader.hpp"
+
+#include "./resource_manager.hpp"
+
+namespace controller_manager
+{
+
+class ResourceStorage
+{
+  static constexpr const char * pkg_name = "hardware_interface";
+
+  static constexpr const char * actuator_interface_name =
+    "hardware_interface::components::ActuatorInterface";
+  static constexpr const char * sensor_interface_name =
+    "hardware_interface::components::SensorInterface";
+  static constexpr const char * system_interface_name =
+    "hardware_interface::components::SystemInterface";
+
+public:
+  ResourceStorage()
+  : actuator_loader_(pkg_name, actuator_interface_name),
+    sensor_loader_(pkg_name, sensor_interface_name),
+    system_loader_(pkg_name, system_interface_name)
+  {}
+
+  ~ResourceStorage() = default;
```

Comment:
Superfluous. Also, we have a ton of them. Can't we just remove them all? None of these are declared virtual or have anything done to them, better leave boilerplate generation to the compiler.